torch/quantization/_equalize.py

```diff
@@ -0,0 +1,89 @@
+import torch
+import torch.nn as nn
+
+def channel_range(tensor, axis =0):
+    ''' finds the range of weights associated with a specific channel
+
+    '''
+    size_of_tensor_dim = len(tensor.size())
+    mins = tensor
+    maxs = tensor
+    for i in range(size_of_tensor_dim - 1, axis, -1):
+        mins = torch.min(mins, i)[0]
+        maxs = torch.max(maxs, i)[0]
+    for i in range(axis):
+        mins = torch.min(mins, 0)[0]
+        maxs = torch.max(maxs, 0)[0]
+    return maxs.sub(mins)
+
+def scaling_channels(tensor1, tensor2, output_axis = 0, input_axis = 1):
+    ''' Given two adjacent tensors', the weights are scaled such that
+    the ranges of the first tensors' output channel are equal to the
+    ranges of the second tensors' input channel
+
+    Note: assumes the input/output channels are the first two axises of the given tensors
+    '''
+    if tensor1.size()[output_axis] != tensor2.size()[input_axis]:
+        raise TypeError("Incompatible tensors")
+
+    output_channel_tensor1 = channel_range(tensor1, output_axis)
+    input_channel_tensor2 = channel_range(tensor2, input_axis)
+
+    # producing scaling factors to applied
+    zero_input = torch.zeros(output_channel_tensor1.size())
```

Comment:
Lets make sure that we add a small constant to input_channel_tensor2 so that we do not divide by zero.